docker-app/qfieldcloud/core/management/commands/extractstoragemetadata.py

```diff
@@ -0,0 +1,133 @@
+import csv
+import logging
+import sys
+from typing import Generator, NamedTuple
+
+import boto3
+import mypy_boto3_s3
+from django.core.management.base import BaseCommand, CommandParser
+from qfieldcloud.core import utils
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Save metadata from S3 storage project files to disk."""
+
+    def add_arguments(self, parser: CommandParser):
+        parser.add_argument(
+            "-o", "--output", type=str, help="Optional: Name of output file"
+        )
+        parser.add_argument(
+            "-s3",
+            "--s3_credentials",
+            type=str,
+            help="""
+            Optional: S3 credentials as comma-separated key=value pairs.
+            Example: -s3 STORAGE_ACCESS_KEY_ID=...,STORAGE_SECRET_ACCESS_KEY=...,...
+        """,
+        )
+
+    def handle(self, *args, **options):
+        config = None
+
+        if options.get("s3_credentials"):
+            config = self.from_shell(options["s3_credentials"])
+            bucket = self.get_s3_bucket(config)
+
+        else:
+            bucket = utils.get_s3_bucket()
+
+        output_name = options.get("output")
+
+        if output_name:
+            name_suffix = output_name.rsplit(".", 1)
+
+            if len(name_suffix) != 2 or name_suffix[1] != "csv":
+                logger.error("The name of the output file should end with '.csv'")
+                sys.exit(1)
+        else:
+            output_name = "s3_storage_files.csv"
```

Comment:
```suggestion
```

No need for that many checks. Your file extension should not be a reason for this command to fail.